wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/util/RecipeDownloader.java

```diff
@@ -58,24 +59,27 @@ public RecipeDownloader(@Named("api.endpoint") String apiEndpoint) {
     public RecipeImpl getRecipe(MachineConfig machineConfig) throws MachineException {
         URL recipeUrl;
         File file = null;
+        final String location = machineConfig.getSource().getLocation();
         try {
-            UriBuilder targetUriBuilder = UriBuilder.fromUri(machineConfig.getSource().getLocation());
+            UriBuilder targetUriBuilder = UriBuilder.fromUri(location);
             // add user token to be able to download user's private recipe
-            if (machineConfig.getSource().getLocation().startsWith(apiEndpoint)) {
+            final String apiEndPointHost = apiEndpoint.getHost();
+            final String host = UriBuilder.fromUri(location).build().getHost();
```

Comment:
yeah, we tried but it does not work, that is why @vparfonov added custom method with redirect.